ui/e2e/src/pages/Panel.ts

```diff
@@ -43,6 +46,27 @@ export class Panel {
     // The classname selector here is not ideal, but it's all that is available
     // because this lives deeper in another library.
     this.resizeHandle = this.container.locator('..').locator('.react-resizable-handle');
+
+    this.figure = this.container.getByRole('figure');
+    this.loader = this.container.getByLabel('Loading');
+  }
+
+  async isLoaded() {
+    // Wait for the figure to have at least one visible child that is not the loader.
+    await expect(async () => {
+      expect(this.figure).not.toContain(this.loader);
+
+      const figureChildren = this.figure.locator('*:visible');
+
+      expect(await figureChildren.count()).toBeGreaterThan(0);
+    }).toPass();
+
+    // Trial a click to wait for the figure to be stable. Replace this with a
+    // baked in "stable" check when one is available.
+    // https://github.com/microsoft/playwright/issues/15195#issuecomment-1176370571
```

Comment:
There's an underlying "stable" check that is really useful before taking snapshots, but it isn't exposed directly, so calling a "trial" click (i.e. a test if the element is clickable according to playwright) is a workaround to get that same check.